pkg/network/bridge_vlan_controller.go

```diff
@@ -39,35 +46,109 @@ func Register(ctx context.Context, apply apply.Apply, setting harvcontroller.Set
 }
 
 func (c *BridgeVLANController) OnChange(key string, setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
-	if setting == nil {
-		return nil, nil
+	networkSetting, err := NewNetworkSetting(setting)
+	if err != nil {
+		return nil, fmt.Errorf("get network setting failed, error: %w, setting: %+v", err, setting)
 	}
-	if setting.Value == "" || key != networkSettingsName {
-		return setting, nil
+
+	if err := onChangeReconcile(br, networkSetting); err != nil {
+		return nil, err
 	}
-	logrus.Printf("harvester network setting configured to: %s", setting.Value)
 
-	// TODO, config/re-config the bridge
-	// 1. convert setting to networSetting struct, return error if is invalid
+	// update configured nic
+	networkSetting.ConfiguredNIC = networkSetting.NIC
+	return c.updateNetworkSetting(networkSetting, setting)
+}
 
-	// 2. if has configured-bridge, reset the bridge and configured NIC first
+func (c *BridgeVLANController) OnRemove(key string, setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
+	// TODO, bridge VLAN setting is not suppose to be removed, therefore we can add a backup setting on delete?
+	networkSetting, err := NewNetworkSetting(setting)
+	if err != nil {
+		return nil, fmt.Errorf("get network setting failed, error;: %w, setting: %+v", err, setting)
+	}
 
-	// 3. set the new configured NIC to the bridge
+	nic, err := bridge.GetLink(networkSetting.NIC)
+	if err != nil {
+		return nil, err
+	}
 
-	// 4. update the status/log of the setting
+	// if return address failed, log and continue
+	if err := br.ReturnAddr(nic); err != nil {
+		klog.Errorf("return address failed, error: %v, NIC: %s", err, networkSetting.NIC)
+	}
+
+	if err := br.Delete(); err != nil {
+		return nil, fmt.Errorf("delete bridge failed, error: %w, bridge: %+v", err, br)
+	}
 
 	return setting, nil
 }
 
-func (c *BridgeVLANController) OnRemove(key string, setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
-	if setting == nil {
-		return nil, nil
-	}
-	if setting.Value == "" || key != networkSettingsName {
-		return setting, nil
+// UpdateNetworkSetting update value field
+func (c *BridgeVLANController) updateNetworkSetting(networkSetting *NetworkSetting, setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
+	bytes, err := json.Marshal(networkSetting)
+	if err != nil {
+		return nil, fmt.Errorf("marshal failed, error: %w, networkSetting: %+v", err, networkSetting)
 	}
 
-	// TODO, bridge VLAN setting is not suppose to be removed, therefore we can add a backup setting on delete?
+	setting.Value = string(bytes)
 
-	return setting, nil
+	return c.settingClient.Update(setting)
```

Comment:
slim this method:
1. do and only do one thing in this method, encodes and return the `networkSetting` JSON string 
2. move the update logic to the onChange method and need to do deepCopy when updating the object.